File: demo/demo-web-admin/src/api/message/interface/message.ts
/**
 * 消息通信相关类型定义
 */


/**
 * 消息通信传输的消息
 * 
 * @author flyan
 */
export interface Message<T> {
    /**
     * 请求/响应服务
     */
    service: string;

    /**
     * 消息类型
     */
    messtype: string;

    /**
     * 消息体
     */
    data: T | null;
}

/**
 * 消息通信接口响应的消息，继承自 {@link Message}，单独定义该消息对象是因为接口
 * 响应的信息并不是所有情况都需要的，例如对于内部消息通信或微服务下的远程调用，它们都有
 * 可能只使用普通消息传递即可。
 * 
 * @author flyan
 */
export interface ResponseMessage<T> {
    /**
     * 错误码
     */
    errcode: number;
    
    /**
     * 错误提示，用户可阅读
     */
    msg: string;

    /**
     * @see {@link Message}
     */
    service: string;
    messtype: string;
    data: T | null;
} 

File: core/src/main/java/com/liuyun/swiftmcweb/core/pojo/Message.java
package com.liuyun.swiftmcweb.core.pojo;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;

/**
 * 消息通信-消息
 *
 * @apiNote 消息通信传输的消息
 * @author flyan
 * @version 1.0
 * @date 10/18/22
 */
//@Schema(description = "消息通信-消息")
@Data
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
public class Message<T> implements Serializable {

    /**
     * 服务
     *
     * @mock    SERVICE
     */
    @Schema(description = "服务", required = true, example = "DEMO")
    protected String service;

    /**
     * 消息类型
     *
     * @mock    coffee
     */
    @Schema(description = "消息类型", required = true, example = "hello_world")
    protected String messtype;

    /**
     * 消息体
     */
    @Schema(description = "消息体", required = true)
    protected T data;

    /**
     * 克隆一个基本消息，不带消息体
     *
     * @return
     */
    public <R> Message<R> baseClone() {
        return new Message<R>().setService(service).setMesstype(messtype);
    }

}
